chromium_src/services/network/cookie_expiry_browsertest.cc

```diff
@@ -0,0 +1,179 @@
+
+#include "base/path_service.h"
+#include "base/strings/stringprintf.h"
+#include "brave/components/constants/brave_paths.h"
+#include "build/build_config.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+
+constexpr base::TimeDelta k4YearsInDays = base::Days(1461);
+
+class CookieExpirationTest : public InProcessBrowserTest {
+ public:
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+    https_server_.reset(new net::EmbeddedTestServer(
+        net::test_server::EmbeddedTestServer::TYPE_HTTPS));
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    https_server_->ServeFilesFromDirectory(test_data_dir);
+    ASSERT_TRUE(https_server_->Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+  }
+
+ protected:
+  std::unique_ptr<net::EmbeddedTestServer> https_server_;
+
+ private:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// Set a cookie with JavaScript.
+void JSDocumentCookieWriteCookie(Browser* browser, std::string age) {
+  bool rv = content::ExecuteScript(
+      browser->tab_strip_model()->GetActiveWebContents(),
+      base::StringPrintf("document.cookie = 'name=Test; %s'", age.c_str()));
+  CHECK(rv);
+}
+
+void JSCookieStoreWriteCookie(Browser* browser, std::string expires_in_ms) {
+  content::EvalJsResult result = content::EvalJs(
+      browser->tab_strip_model()->GetActiveWebContents(),
+      base::StringPrintf("async function doSet() {"
+                         "  await window.cookieStore.set("
+                         "       { name: 'name',"
+                         "         value: 'Good',"
+                         "         expires: Date.now() + %s,"
+                         "          });"
+                         "  window.domAutomationController.send(true);"
+                         "}"
+                         "doSet()",
```

Comment:
no need to do the manual `domAutomationController.send()` stuff. `EvalJs/ExecJs` supports promises and async calls.
https://github.com/chromium/chromium/blob/396799b65ad67a7ea10bbdeb1d75029d297bb520/content/public/test/browser_test_utils.h#L803-L809